app/forms/schools/add_participant_form.rb

```diff
@@ -107,27 +107,21 @@ def current_user
       @current_user ||= User.find_by(id: current_user_id)
     end
 
+    def creators
+      {
+        ect: EarlyCareerTeachers::Create,
+        mentor: Mentors::Create,
+      }
+    end
+
     def save!
-      if type == :ect
-        EarlyCareerTeachers::Create.call(
-          full_name: full_name,
-          email: email,
-          cohort_id: school_cohort.cohort_id,
-          school_id: school_cohort.school_id,
-          mentor_profile_id: mentor&.mentor_profile&.id,
-        )
-      else
-        ActiveRecord::Base.transaction do
-          # TODO: What if email matches but with different name?
-          user = User.find_or_create_by!(full_name: full_name, email: email)
-
-          MentorProfile.create!(
-            user: user,
-            school_id: school_cohort.school_id,
-            cohort_id: school_cohort.cohort_id,
-          )
-        end
-      end
+      creators[type]&.call(
```

Comment:
I think this should be `creators[participant_type]`.  I suspect the error you were having was caused by `type = :self`

Reply:
Yes. I suspect that you are correct. Highlights that there mustn't have been a unit test for that scenario.